docs/generate_py_docs.py

```diff
@@ -0,0 +1,170 @@
+                 'doctitle_xform': doctitle,
+                 'initial_header_level': initial_header_level}
+    parts = core.publish_parts(
+        source=input_string, source_path=source_path,
+        destination_path=destination_path,
+        writer_name='html', settings_overrides=overrides)
+    return parts['body']
+
+
+def html2txt(html: str):
+    html_elements = BeautifulSoup(html, features="html.parser")
+
+    # remove style tags
+    for tag in html_elements.select('style'):
+        tag.decompose()
+
+    for tag in html_elements.select('td'):
+        # remove non-breaking spaces
+        if tag.text == u'\xa0':
+            tag.decompose()
+        else:
+            # add new line and indent for the cells inside a table
+            if 'class' not in tag.attrs:
+                tag.insert_before('\n      ')
+
+    # substitute double backticks with a single backtick
+    for tag in html_elements.select('tt'):
+        tag.insert_before('`')
+        tag.insert_after('`')
+
+    # make the option's text inside the option list bold
+    for tag in html_elements.select('span'):
+        if 'class' in tag.attrs:
+            if ' '.join(tag.attrs['class']) == 'option':
+                tag.insert_before('<b>')
+                tag.insert_after('</b>')
+
+    # keep bold text bold
+    for tag in html_elements.select('strong'):
+        tag.insert_before('<b>')
+        tag.insert_after('</b>')
+
+    # add new lines after the paragraph tag
+    for tag in html_elements.select('p'):
+        if 'class' in tag.attrs:
+            if ' '.join(tag.attrs['class']) == 'first':
+                tag.insert_before(' ')
+            else:
+                tag.insert_before('\n')
+        else:
+            tag.insert_before('\n\n')
+
+    # format elements in the `Option:` section by adding new line and indent
+    for tag in html_elements.select('kbd'):
+        tag.insert_before('\n   ')
+
+    # add indent to the description list
+    for tag in html_elements.select('dt'):
+        tag.string = '\n' + indent(tag.text.strip(), '   ')
+
+    # add indent to the nested description list
+    for tag in html_elements.select('dd'):
+        tag.string = '\n' + indent(tag.text, '      ')
+
+    # add indent and bullet points to the bulleted list
+    for tag in html_elements.select('li'):
+        if tag.parent.parent.name == 'li':  # deal with nested lists
+            tag.insert_before('\n   - ')
+        else:
+            tag.insert_before('\n* ')
+
+    # add new line before included NOTE class of rst documentation, example: github_api_access.rst
+    for tag in html_elements.select('div'):
+        if 'class' in tag.attrs:
+            if 'admonition note' in ' '.join(tag.attrs['class']):
+                tag.insert_before('\n')
+
+    # format code examples
+    for tag in html_elements.select('pre'):
+        if 'class' in tag.attrs:
+            if ' '.join(tag.attrs['class']) in ['code shell literal-block', 'code awk literal-block']:
```

Comment:
Would it be possible not to rely on specific languages like `shell` or `awk` (as they were selected somewhat randomly IIRC), but apply the transformation to `pre`s with `code literal-block` classes?

Reply:
Yes, will do